src/features/DebugSession.ts

```diff
@@ -40,4 +40,109 @@ export class DebugSessionFeature implements IFeature {
     }
-}
\ No newline at end of file
+}
+
+interface ProcessItem extends vscode.QuickPickItem {
+	pid: string;	// payload for the QuickPick UI
+}
+
+interface PSHostProcessInfo {
+    processName: string;
+    processId: string;
+    appDomainName: string;
+    mainWindowTitle: string;
+}
+
+namespace GetPSHostProcessesRequest {
+    export const type: RequestType<any, GetPSHostProcessesResponseBody, string> =
+        { get method() { return 'powerShell/getPSHostProcesses'; } };
+}
+
+interface GetPSHostProcessesResponseBody {
+    hostProcesses: PSHostProcessInfo[];
+}
+
+export class PickPSHostProcessFeature implements IFeature {
+
+    private command: vscode.Disposable;
+    private languageClient: LanguageClient;
+    private waitingForClientToken: vscode.CancellationTokenSource;
+
+    constructor() {
+        this.command =
+            vscode.commands.registerCommand('PowerShell.PickPSHostProcess', () => {
+
+                if (!this.languageClient && !this.waitingForClientToken) {
+
+                    // If PowerShell isn't finished loading yet, show a loading message
+                    // until the LanguageClient is passed on to us
+                    this.waitingForClientToken = new vscode.CancellationTokenSource();
+                    vscode.window
+                        .showQuickPick(
+                            ["Cancel"],
+                            { placeHolder: "Select PowerShell Host Process to attach to: Please wait, starting PowerShell..." },
+                            this.waitingForClientToken.token)
+                        .then(response => { if (response === "Cancel") { this.clearWaitingToken(); } });
+
+                    // Cancel the loading prompt after 60 seconds
+                    setTimeout(() => {
+                            if (this.waitingForClientToken) {
+                                this.clearWaitingToken();
+
+                                vscode.window.showErrorMessage(
+                                    "Select PowerShell Host Process to attach to: PowerShell session took too long to start.");
+                            }
+                        }, 60000);
+                }
+                else {
+                    return this.pickPSHostProcess();
+                }
+            });
+    }
+
+    public setLanguageClient(languageClient: LanguageClient) {
+        this.languageClient = languageClient;
+
+        if (this.waitingForClientToken) {
+            this.clearWaitingToken();
+            return this.pickPSHostProcess();
+        }
+    }
+
+    public dispose() {
+        this.command.dispose();
+    }
+
+    // In node, the function returned a Promise<string> not sure about "Thenable<string>"
+	private pickPSHostProcess(): Thenable<string> {
+		return this.languageClient.sendRequest(GetPSHostProcessesRequest.type, null).then(hostProcesses => {
+			var items: ProcessItem[] = [];
+
+			for(var p in hostProcesses) {
```

Comment:
Maybe if the length of hostProcesses is 0 we should use `vscode.window.showInformationMessage` to inform the user that there are no host processes to choose from?